adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,994 @@
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (1)
+
+template <typename SrcT, int SrcDims, access_mode SrcMode,
+          typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (1)
+
+}
+----
+
+Copies between two device accessors.
+
+_Constraints_:
+
+* [code]#SrcT# and [code]#DestT# is <<device-copyable>>.
+* [code]#SrcMode# is [code]#access_mode::read# or
+  [code]#access_mode::read_write#.
+* [code]#DestMode# is [code]#access_mode::write# or
+  [code]#access_mode::read_write#.
+
+_Effects (1)_: Equivalent to [code]#h.copy(src, dest)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { copy(h, src,
+dest); })#.
+
+_Throws_: A synchronous [code]#exception# with the [code]#errc::invalid# error
+code if [code]#dest.get_count() < src.get_count()#.
+
+'''
+
+.[apititle]#memset#
+[source,role=synopsis,id=api:memset]
+----
+namespace sycl::khr {
+
+void memset(sycl::handler& h, void* ptr, int value, size_t numBytes); (1)
+
+void memset(sycl::queue q, void* ptr, int value, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.memset(ptr, value, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { memset(h, value,
+numBytes); })#.
+
+'''
+
+.[apititle]#fill#
+[source,role=synopsis,id=api:fill]
+----
+namespace sycl::khr {
+
+template <typename T>
+void fill(sycl::handler& h, T* ptr, const T& pattern, size_t count); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::handler& h,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);                                             (2)
+
+template <typename T>
+void fill(sycl::queue q, T* ptr, const T& pattern, size_t count);    (3)
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::queue q,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);                                             (4)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.fill(ptr, pattern, count)#.
+
+_Effects (2)_: Equivalent to [code]#h.fill(dest, src)#.
+
+_Effects (3)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, ptr,
+pattern, count); })#.
+
+_Effects (4)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, dest,
+src); })#.
+
+'''
+
+.[apititle]#update_host#
+[source,role=synopsis,id=api:update_host]
+----
+namespace sycl::khr {
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::handler& h, accessor<T, Dims, Mode, target::device> acc); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::queue q, accessor<T, Dims, Mode, target::device> acc);    (2)
+
+}
+----
+_Constraints_: [code]#T# is <<device-copyable>>.
```

Comment:
We do not need to say this here because `T` is the element type of `accessor` and `accessor` already has a requirement that its element type is device copyable.  Therefore, it would not be possible to even construct an `accessor` with a non-device-copyable element type.

Reply:
Good point.  Removed in a9bdc102f37d54ae1cbadb103b67f4c992f7df45.